app/models.py

```diff
@@ -244,8 +252,12 @@ def update(self, input_host, update_system_profile=False):
         if update_system_profile:
             self.update_system_profile(input_host.system_profile_facts)
 
-        self._update_stale_timestamp(input_host.stale_timestamp, input_host.reporter)
-        self._update_per_reporter_staleness(input_host.stale_timestamp, input_host.reporter)
+        if self.system_profile_facts and self.system_profile_facts.get("host_type") == "edge":
+            self._update_stale_timestamp(EDGE_HOST_STALE_TIMESTAMP, input_host.reporter)
+            self._update_per_reporter_staleness(EDGE_HOST_STALE_TIMESTAMP, input_host.reporter)
+        else:
+            self._update_stale_timestamp(input_host.stale_timestamp, input_host.reporter)
+            self._update_per_reporter_staleness(input_host.stale_timestamp, input_host.reporter)
```

Comment:
Since it's the same two calls either way could we just conditionally assign the timestamp to avoid repeating the calls? It will make it easier if we need to add other timestamp dependent calls later, or if the interface for either function changes.

Also switched to a optional chaining pattern to simplify/shorten the condition. LMK what you think of it. I like that it keeps you from having to check each part of the chain individually, but the syntax isn't as nice as say JS's `?.`. I don't personally think it hurts readability, especially if we start using it in more places, but that doesn't mean it doesn't.

It will just return None if any part of the chain fails, since an empty dict will never have whatever member you're asking for.

```suggestion
        stale_timestamp = input_host.stale_timestamp
        
        if self.get(system_profile, {}).get("host_type") == "edge":
            stale_timestamp = EDGE_HOST_STALE_TIMESTAMP
   
        self._update_stale_timestamp(stale_timestamp, input_host.reporter)
        self._update_per_reporter_staleness(stale_timestamp, input_host.reporter)
```